crates/ruff_python_ast/src/nodes.rs

```diff
@@ -2987,8 +2987,8 @@ pub struct ParameterWithDefault {
 #[derive(Clone, Debug, PartialEq)]
 pub struct Arguments {
     pub range: TextRange,
-    pub args: Vec<Expr>,
-    pub keywords: Vec<Keyword>,
+    pub args: Box<[Expr]>,
+    pub keywords: Box<[Keyword]>,
```

Comment:
We could make these `Option<Box<...>>` to avoid allocating when empty...

Reply:
It should be possible to use `Option` without increasing the struct size, and avoiding the allocation could be nice. although It doesn't seem that the empty slice allocates 

https://godbolt.org/z/E9cxqE9P3 CC: @BurntSushi

Reply:
Gasp...

Reply:
Yeah that is interesting. I actually couldn't find how this was being done in std specifically. So it could just be an optimization in the code generator somewhere. Or perhaps I missed a call to [`dangling`](https://doc.rust-lang.org/std/ptr/struct.NonNull.html#method.dangling).